### PR TITLE
Lab4 numeronym

### PR DESCRIPTION
**Pull request by vinilapeddireddy**
Fixes #67

#### Changes proposed in this PR:

- Implemented numeronyms function for its input strings
- Add unit test for numeronyms function

### CODE REVIEW

File: 04_numeronym/vinilapeddireddy/main.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	"fmt"
+	"io"
+	"os"
+	"strings"
+)
+
+var out io.Writer = os.Stdout
+
+func numeronyms(s ...string) []string {
+	var numeronyms []string
```
Comment (juliaogris) [inline 04_numeronym/vinilapeddireddy/main.go]: This is fine, but if you really care about runtime you could optimise by preallocating the size of `numeronyms`:
```go
numeronyms := make([]string, len(vals))
for i, val := range vals {
		numeronyms[i] =  numeronym(val)
}
```

File: 04_numeronym/vinilapeddireddy/main_test.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	"github.com/stretchr/testify/require"
+)
+
+var tests = []struct {
+	inputStrings  []string
+	outputStrings []string
+}{
+	{[]string{}, nil},
+	{[]string{"accessibility", "Kubernetes", "abc", "charecters"}, []string{"a11y", "K8s", "abc", "c8s"}},
+	{[]string{"!!&^#*&+"}, []string{"!6+"}},
```
Comment (juliaogris) [inline 04_numeronym/vinilapeddireddy/main_test.go:16]: Nice, could we have a test with spacd and unicode as well please?

**Review by juliaogris — COMMENTED** [review]
Pretty close!

Please edit your commits to follow [recommendations](https://chris.beams.io/posts/git-commit/) 
- Capitalize the subject line 
- Use the imperative mood in the subject line 
- Fold fixup commits into previous ones with `git rebase -i HASH`
    I can give a hand with this if you want

Also, I've added some slides for doing this: https://gotraining.xyz/gotraining.slide#29

**Review by juliaogris — COMMENTED** [review]
Please edit your commits to follow [recommendations](https://chris.beams.io/posts/git-commit/)

-   Capitalize the subject line
-   Use the imperative mood in the subject line
-   Fold fixup commits into previous ones with `git rebase -i HASH`
    See [slides](https://gotraining.xyz/gotraining.slide#29).

Other than that only minor things.

File: 04_numeronym/vinilapeddireddy/main_test.go (hunk 2 above)
Comment (vinilapeddireddy) [reply 04_numeronym/vinilapeddireddy/main_test.go:16]: Done

File: 04_numeronym/vinilapeddireddy/main.go (hunk 1 above)
Comment (vinilapeddireddy) [reply 04_numeronym/vinilapeddireddy/main.go]: Done

**Comment by juliaogris** [comment]
Hi,
There has been  no activity in the last 20 days so I will close this PR.
Feel free to open it again if you have something to add.
Thanks!
Julia

**Pull request closed without merging**